feature/topics/src/main/kotlin/uk/govuk/app/topics/ui/TopicScreen.kt

```diff
@@ -1,27 +1,226 @@
+import uk.govuk.app.design.ui.theme.GovUkTheme
+import uk.govuk.app.topics.R
+import uk.govuk.app.topics.TopicViewModel
+import uk.govuk.app.topics.ui.model.TopicUi
 
 @Composable
 internal fun TopicRoute(
-    title: String,
+    onBack: () -> Unit,
+    onExternalLink: (url: String) -> Unit,
+    onStepByStepSeeAll: () -> Unit,
+    onSubtopic: (ref: String) -> Unit,
     modifier: Modifier = Modifier
 ) {
+    val viewModel: TopicViewModel = hiltViewModel()
+    val topic by viewModel.topic.collectAsState()
+
     TopicScreen(
-        title = title,
+        topic = topic,
+        onPageView = { title -> viewModel.onPageView(title) },
+        onBack = onBack,
+        onExternalLink = { section, text, url ->
+            viewModel.onContentClick(
+                section = section,
+                text = text,
+                url = url
+            )
+            onExternalLink(url)
+        },
+        onStepByStepSeeAll = { section, text ->
+            onStepByStepSeeAll()
+            viewModel.onSeeAllClick(
+                section = section,
+                text = text
+            )
+        },
+        onSubtopic = { text, ref ->
+            viewModel.onSubtopicClick(text)
+            onSubtopic(ref)
+        },
         modifier = modifier
     )
 }
 
 @Composable
 private fun TopicScreen(
-    title: String,
+    topic: TopicUi?,
+    onPageView: (String) -> Unit,
+    onBack: () -> Unit,
+    onExternalLink: (section: String, text: String, url: String) -> Unit,
+    onStepByStepSeeAll: (section: String, text: String) -> Unit,
+    onSubtopic: (text: String, ref: String) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    Title1BoldLabel(
-        text = title,
-        modifier = modifier
-    )
-}
\ No newline at end of file
+    Column(modifier.fillMaxSize()) {
+        if (topic != null) {
+            LaunchedEffect(Unit) {
+                onPageView(topic.title)
+            }
+
+            ChildPageHeader(
+                text = topic.title,
+                onBack = onBack
+            )
+
+            LazyColumn(Modifier.padding(horizontal = GovUkTheme.spacing.medium)) {
+                if (topic.description != null) {
+                    item {
+                        BodyRegularLabel(topic.description)
+                    }
+
+                    item {
+                        MediumVerticalSpacer()
+                    }
+                }
+
+                popularPages(
+                    popularPages = topic.popularPages,
+                    onClick = onExternalLink
+                )
+
+                stepBySteps(
+                    stepBySteps = topic.stepBySteps,
+                    displaySeeAll = topic.displayStepByStepSeeAll,
+                    onClick = onExternalLink,
+                    onSeeAll = onStepByStepSeeAll
+                )
+
+                subtopics(
+                    subtopics = topic.subtopics,
+                    onClick = onSubtopic
+                )
```

Comment:
Maybe not a question for you or here. But, I'm wondering what, if anything, are we doing about cyclic loops in the Topics hierarchy 🤔  While clicking through the UI at random I noticed that I was being led from one top level, into another top-level's sub-topics structure. I found this oddly disconcerting as it felt like I'd literally gone off-topic 😕 I imagine some of our users could get very confused by this.

Reply:
Yeh, I noticed that you can get pretty deep into it. Not sure if it's been considered, I'll raise it with Ryan.

Reply:
Just raised it in post amigo, they are aware of it as a known risk, it's down to the content team to manage it. On iOS they have a work around where if you hit the home tab it boots you back up to the home page. Will probably implement the same thing on Android.

Reply:
I don't think this is something we can or should try to solve within the app. I feel it needs to be resolved at the Topic hierarchy level.